tests/integration/cmd_pref_config_test.go

```diff
@@ -103,6 +111,15 @@ var _ = Describe("odo preference and config command tests", func() {
 			helper.CmdShouldFail("odo", "preference", "set", "-f", "experimental", "invalid-value")
 			helper.CmdShouldFail("odo", "preference", "set", "-f", "updatenotification", "invalid-value")
 		})
+
+		It("should show json output", func() {
+			prefJSONOutput, err := helper.Unindented(helper.CmdShouldPass("odo", "preference", "view", "-o", "json"))
+			Expect(err).Should(BeNil())
+			expected, err := helper.Unindented(`{"kind":"PreferenceList","apiVersion":"odo.dev/v1alpha1","items":[{"Name":"UpdateNotification","Value":null,"Default":true,"Type":"bool","Description":"Flag to control if an update notification is shown or not (Default: true)"},{"Name":"NamePrefix","Value":null,"Default":"","Type":"string","Description":"Use this value to set a default name prefix (Default: current directory name)"},{"Name":"Timeout","Value":null,"Default":1,"Type":"int","Description":"Timeout (in seconds) for OpenShift server connection check (Default: 1)"},{"Name":"BuildTimeout","Value":null,"Default":300,"Type":"int","Description":"BuildTimeout (in seconds) for waiting for a build of the git component to complete (Default: 300)"},{"Name":"PushTimeout","Value":null,"Default":240,"Type":"int","Description":"PushTimeout (in seconds) for waiting for a Pod to come up (Default: 240)"},{"Name":"Experimental","Value":null,"Default":false,"Type":"bool","Description":"Set this value to true to expose features in development/experimental mode"},{"Name":"PushTarget","Value":null,"Default":"kube","Type":"string","Description":"Set this value to 'kube' or 'docker' to tell odo where to push applications to. (Default: kube)"}]}`)
```

Comment:
TBH its really hard for me to understand. @girishramnani are all the fields constant forever? I am confused because its seems such a big json format, so I am worried what if any of the fields dynamically changes.

Holding the pr for now.
/hold

Reply:
Please observe that we are creating a new preference file per case which means this representation won’t change and get affected by other tests

Reply:
Can't we short this expected json? Is there any way?

Reply:
I have opened an issue https://github.com/openshift/odo/issues/3823 for this because its looks really clumsy and confusing when you see the whole json like this.